core/src/avm2/globals/flash/ui/Keyboard.as

```diff
@@ -0,0 +1,366 @@
+        public static const F3: uint = 114;
+        public static const F4: uint = 115;
+        public static const F5: uint = 116;
+        public static const F6: uint = 117;
+        public static const F7: uint = 118;
+        public static const F8: uint = 119;
+        public static const F9: uint = 120;
+        public static const F10: uint = 121;
+        public static const F11: uint = 122;
+        public static const F12: uint = 123;
+        public static const F13: uint = 124;
+        public static const F14: uint = 125;
+        public static const F15: uint = 126;
+        public static const SEMICOLON: uint = 186;
+        public static const EQUAL: uint = 187;
+        public static const COMMA: uint = 188;
+        public static const MINUS: uint = 189;
+        public static const PERIOD: uint = 190;
+        public static const SLASH: uint = 191;
+        public static const BACKQUOTE: uint = 192;
+        public static const LEFTBRACKET: uint = 219;
+        public static const BACKSLASH: uint = 220;
+        public static const RIGHTBRACKET: uint = 221;
+        public static const QUOTE: uint = 222;
+
+        public static const RED: uint = 0x01000000;
+        public static const GREEN: uint = 0x01000001;
+        public static const YELLOW: uint = 0x01000002;
+        public static const BLUE: uint = 0x01000003;
+        public static const CHANNEL_UP: uint = 0x01000004;
+        public static const CHANNEL_DOWN: uint = 0x01000005;
+        public static const RECORD: uint = 0x01000006;
+        public static const PLAY: uint = 0x01000007;
+        public static const PAUSE: uint = 0x01000008;
+        public static const STOP: uint = 0x01000009;
+        public static const FAST_FORWARD: uint = 0x0100000A;
+        public static const REWIND: uint = 0x0100000B;
+        public static const SKIP_FORWARD: uint = 0x0100000C;
+        public static const SKIP_BACKWARD: uint = 0x0100000D;
+        public static const NEXT: uint = 0x0100000E;
+        public static const PREVIOUS: uint = 0x0100000F;
+        public static const LIVE: uint = 0x01000010;
+        public static const LAST: uint = 0x01000011;
+        public static const MENU: uint = 0x01000012;
+        public static const INFO: uint = 0x01000013;
+        public static const GUIDE: uint = 0x01000014;
+        public static const EXIT: uint = 0x01000015;
+        public static const BACK: uint = 0x01000016;
+        public static const AUDIO: uint = 0x01000017;
+        public static const SUBTITLE: uint = 0x01000018;
+        public static const DVR: uint = 0x01000019;
+        public static const VOD: uint = 0x0100001A;
+        public static const INPUT: uint = 0x0100001B;
+        public static const SETUP: uint = 0x0100001C;
+        public static const HELP: uint = 0x0100001D;
+        public static const MASTER_SHELL: uint = 0x0100001E;
+        public static const SEARCH: uint = 0x0100001F;
+        public static const PLAY_PAUSE: uint = 0x01000020;
```

Comment:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#PLAY_PAUSE

"This constant is supported on AIR for TV. On other platforms, its value is undefined."

Not sure it matters though.

Reply:
Strangely, the documentation says the same thing for all the following constants, but only PLAY_PAUSE is marked with the AIR-only symbol:

- [AUDIO](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#AUDIO)
- [BACK](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#BACK)
- [BLUE](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#BLUE)
- [CHANNEL_DOWN](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#CHANNEL_DOWN)
- [CHANNEL_UP](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#CHANNEL_UP)
- [DVR](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#DVR)
- [EXIT](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#EXIT)
- [FAST_FORWARD](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#FAST_FORWARD)
- [GREEN](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#GREEN)
- [GUIDE](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#GUIDE)
- [HELP](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#HELP)
- [INFO](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#INFO)
- [INPUT](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#INPUT)
- [LAST](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#LAST)
- [LIVE](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#LIVE)
- [MASTER_SHELL](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#MASTER_SHELL)
- [MENU](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#MENU)
- [NEXT](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#NEXT)
- [PAUSE](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#PAUSE)
- [PLAY](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#PLAY)
- [PREVIOUS](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#PREVIOUS)
- [RECORD](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#RECORD)
- [RED](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#RED)
- [REWIND](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#REWIND)
- [SEARCH](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#SEARCH)
- [SETUP](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#SETUP)
- [SKIP_BACKWARD](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#SKIP_BACKWARD)
- [SKIP_FORWARD](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#SKIP_FORWARD)
- [STOP](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#STOP)
- [SUBTITLE](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#SUBTITLE)
- [VOD](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#VOD)
- [YELLOW](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/ui/Keyboard.html#YELLOW)

Reply:
Nice catch, I removed them all.